packages/icrc-ledger-types/src/icrc21/lib.rs

```diff
@@ -169,13 +169,19 @@ impl ConsentMessageBuilder {
                 if from_account.owner == Principal::anonymous() {
                     message.push_str(&format!(
                         "\n\n**From Subaccount:**\n{}",
-                        from_account.to_string().split('.').last().ok_or(
-                            Icrc21Error::GenericError {
-                                error_code: Nat::from(500u64),
-                                description: "Sender Subaccount has an unexpected format."
-                                    .to_owned(),
-                            }
-                        )?
+                        match from_account.subaccount {
```

Comment:
Maybe extract that logic and move it to `Account`?

Reply:
Account is part of the icrc1 standard and the way that accounts are displayed here is not consistent with the icrc1 standard, it is a custom solution for the ICRC21 standard. This is why logically it does not fit where Account is defined.

Reply:
In that case it might still be worthwhile to extract a function to remove the duplication?

Reply:
Done